.github/workflows/kyber.yml

```
@@ -0,0 +1,105 @@
+name: Kyber Tests
+
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  build_liboqs:
+    name: Build liboqs
+    runs-on: ubuntu-latest
+    timeout-minutes: 4
+    steps:
+      - name: Checking cache for liboqs
+        uses: actions/cache/restore@v4
+        id: cache-liboqs
+        with:
+          path: opt/liboqs/
+          key: wolfssh-liboqs-0.10.0-${{ runner.os }}
+          lookup-only: true
+
+      - name: Checkout liboqs
+        if: steps.cache-liboqs.outputs.cache-hit != 'true'
+        uses: actions/checkout@v4
+        with:
+          repository: open-quantum-safe/liboqs
+          ref: 0.10.0
+          path: liboqs
+
+      - name: Build and install liboqs
+        if: steps.cache-liboqs.outputs.cache-hit != 'true'
+        working-directory: liboqs
+        run: |
+          mkdir build
+          cd build
+          cmake -DCMAKE_INSTALL_PREFIX=${{ github.workspace }}/opt/liboqs -DOQS_MINIMAL_BUILD=KEM_kyber_512 -DOQS_USE_OPENSSL=0 ..
+          make
+          make install
+
+      - name: Stash liboqs in cache
+        if: steps.cache-liboqs.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          path: opt/liboqs/
+          key: wolfssh-liboqs-0.10.0-${{ runner.os }}
```

You can just use the `actions/cache` to check and save the cache. It will automatically update the cache if the path is present.

```suggestion
      - name: Checking cache for liboqs
        uses: actions/cache@v4
        id: cache-liboqs
        with:
          path: opt/liboqs/
          key: wolfssh-liboqs-0.10.0-${{ runner.os }}
          lookup-only: true

      - name: Checkout liboqs
        if: steps.cache-liboqs.outputs.cache-hit != 'true'
        uses: actions/checkout@v4
        with:
          repository: open-quantum-safe/liboqs
          ref: 0.10.0
          path: liboqs

      - name: Build and install liboqs
        if: steps.cache-liboqs.outputs.cache-hit != 'true'
        working-directory: liboqs
        run: |
          mkdir build
          cd build
          cmake -DCMAKE_INSTALL_PREFIX=${{ github.workspace }}/opt/liboqs -DOQS_MINIMAL_BUILD=KEM_kyber_512 -DOQS_USE_OPENSSL=0 ..
          make
          make install
```